configuration/src/main/java/pl/edu/uj/kimage/configuration/SettingName.java

```diff
@@ -0,0 +1,5 @@
+package pl.edu.uj.kimage.configuration;
+
+public enum SettingName {
+    NUMBER_OF_THREADS
```

Comment:
If you really want to follow YAGNI rule then this enum is an overkill. You have only one property now.